libcraft/items/src/inventory_slot.rs

```diff
@@ -99,31 +104,34 @@ impl InventorySlot {
     }
 
     /// Returns the number of items stored in the inventory slot.
-    pub fn count(&self) -> u32 {
+    #[must_use]
+    pub const fn count(&self) -> u32 {
         match self {
-            InventorySlot::Filled(stack) => stack.count(),
-            InventorySlot::Empty => 0,
+            Self::Filled(stack) => stack.count(),
```

Comment:
Maybe implement `deref` into an `Option` and then use `map` and `unwrap_or_default`?

Reply:
I think that would unneccesarily reduce code readability for little to no reason.

Reply:
readability? `self.map(ItemStack::count).unwrap_or_default()`. Pretty readable in my opinion, take the slot and map it to the itemstack count or return default. Instead of default you could just write `unwrap_or(0)`.